cstrml/market/src/lib.rs

```diff
@@ -773,6 +768,32 @@ impl<T: Trait> Module<T> {
         price.checked_mul(&<T::CurrencyToBalance
         as Convert<u64, BalanceOf<T>>>::convert(rounded_file_size * (duration as u64)))
     }
+
+    fn get_current_block_number() -> BlockNumber {
+        let current_block_number = <system::Module<T>>::block_number();
+        TryInto::<u32>::try_into(current_block_number).ok().unwrap()
+    }
+
+    fn get_reward_ratio(order_id: &T::Hash) -> Perbill {
+        let mut reward_ratio: Perbill = Perbill::one();
+        if let Some(punishment) = Self::merchant_punishments(order_id) {
+            let punishment_ratio: f64 = punishment.success as f64 / (punishment.success + punishment.failed) as f64;
+            if punishment_ratio >= 0.99 {
+                reward_ratio = Perbill::one();
+            } else if punishment_ratio >= 0.98 {
+                reward_ratio = Perbill::from_percent(95);
+            } else if punishment_ratio >= 0.95 {
+                reward_ratio = Perbill::from_percent(90);
+            } else if punishment_ratio >= 0.90 {
+                reward_ratio = Perbill::from_percent(80);
+            } else if punishment_ratio >= 0.85 {
+                reward_ratio = Perbill::from_percent(50);
+            } else {
```

Comment:
if `ratio < 0.85`, sOrder should close immediately, and the remaining amount should turns back to user.

Reply:
Yes. That's the expected behavior of current implementation.